code/client/src/components/AddressInput.jsx

```diff
@@ -0,0 +1,103 @@
+import { SearchOutlined } from '@ant-design/icons'
+import { Select } from 'antd'
+import React, { useCallback } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import walletActions from '../state/modules/wallet/actions'
+import util, { useWindowDimensions } from '../util'
+
+/**
+ * Renders address input that provides type ahead search for any known ONE addresses.
+ * Known addresses are addresses that have been entered by user for at least once.
+ * If an address is entered in none one1... format, the address will be converted and saved as one1... format.
```

Comment:
see below comment on always using hex format in state

Reply:
Done.